File: backend/src/main/java/com/example/spotify_app/model/Track/Track.java
package com.example.spotify_app.model.Track;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import com.example.spotify_app.model.common.ExternalUrls;
import com.example.spotify_app.model.Album.Album;
import com.example.spotify_app.model.Artist.Artist;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Track {

    @JsonProperty("album")
    private Album album;

    @JsonProperty("artists")
    private Artist[] artists;

    @JsonProperty("disc_number")
    private int discNumber;

    @JsonProperty("duration_ms")
    private int durationMs;

    @JsonProperty("external_urls")
    private ExternalUrls externalUrls;

    @JsonProperty("href")
    private String href;

    @JsonProperty("id")
    private String id;

    @JsonProperty("is_playable")
    private Boolean isPlayable;

    @JsonProperty("name")
    private String name;

    @JsonProperty("popularity")
    private int popularity;

    @JsonProperty("preview_url")
    private String previewUrl;

    @JsonProperty("track_number")
    private int trackNumber;

    @JsonProperty("type")
    private String type;

    @JsonProperty("uri")
    private String uri;

    @JsonProperty("explicit")
    private Boolean explicit;
}

File: frontend/src/types/Track/Track.ts
import type {
    ExternalUrls,
    Image
} from '../Common';
import type { Artist } from '../Artist/Artist';

export interface TrackAlbum {
    id: string;
    name: string;
    album_type: string;
    total_tracks: number;
    href: string;
    images: Image[];
    external_urls: ExternalUrls;
    release_date: string;
    type: string;
    uri: string;
    artists: Artist[];
}

export interface Track {
    album: TrackAlbum;
    artists: Artist[];
    disc_number: number;
    duration_ms: number;
    external_urls: ExternalUrls;
    href: string;
    id: string;
    is_playable?: boolean;
    name: string;
    popularity: number;
    preview_url: string | null;
    track_number: number;
    type: string;
    uri: string;
    explicit?: boolean;
}
